pleskdistup/actions/spamassassin.py

```diff
@@ -31,3 +33,74 @@ def _post_action(self) -> action.ActionResult:
     def _revert_action(self) -> action.ActionResult:
         os.unlink(self.spamassasin_backup_path)
         return action.ActionResult()
+
+
+class HandleUpdatedSpamassassinConfig(action.ActiveAction):
+    # Make sure the trick is preformed before any call of 'systemctl daemon-reload'
+    # because we change spamassassin.service configuration in scope of this action.
+    def __init__(self):
+        self.name = "handle spamassassin configuration update"
+
+    def _is_required(self) -> bool:
+        return packages.is_package_installed("psa-spamassassin")
+
+    def _prepare_action(self) -> action.ActionResult:
+        util.logged_check_call(["/usr/bin/systemctl", "stop", "spamassassin.service"])
+        util.logged_check_call(["/usr/bin/systemctl", "disable", "spamassassin.service"])
+        return action.ActionResult()
+
+    def _post_action(self) -> action.ActionResult:
+        util.logged_check_call(["/usr/sbin/plesk", "sbin", "spammng", "--enable"])
+        util.logged_check_call(["/usr/sbin/plesk", "sbin", "spammng", "--update", "--enable-server-configs", "--enable-user-configs"])
+
+        util.logged_check_call(["/usr/bin/systemctl", "daemon-reload"])
+        util.logged_check_call(["/usr/bin/systemctl", "enable", "spamassassin.service"])
+
+        # TODO. Following action is not supported on deb-based system. Actually it will be just skipped.
+        # So if you are going to use the action on deb-based, you should be ready there will be no .rpmnew
+        # things or even file here (obviously).
+        if dist.get_distro().rhel_based():
```

Comment:
It depends on changes in the spamassasin package. If I recall correctly, we should make these changes because we need to add plesk configurations to the new configuration file.
This is the main purpose of the action. Therefore, if we do not need to do this, we should not use the action at all.